central/deployment/store/postgres/search_comparison_test.go

```diff
@@ -0,0 +1,170 @@
+package postgres
+
+import (
+	"context"
+	"sort"
+	"testing"
+
+	"github.com/golang/mock/gomock"
+	imageDataStore "github.com/stackrox/rox/central/image/datastore"
+	v1 "github.com/stackrox/rox/generated/api/v1"
+	"github.com/stackrox/rox/generated/storage"
+	"github.com/stackrox/rox/pkg/env"
+	"github.com/stackrox/rox/pkg/fixtures"
+	"github.com/stackrox/rox/pkg/postgres/pgtest"
+	pkgSchema "github.com/stackrox/rox/pkg/postgres/schema"
+	"github.com/stackrox/rox/pkg/sac"
+	"github.com/stackrox/rox/pkg/search"
+	"github.com/stackrox/rox/pkg/search/predicate"
+	"github.com/stretchr/testify/assert"
+	"github.com/stretchr/testify/require"
+	"github.com/stretchr/testify/suite"
+)
+
+var (
+	ctx = sac.WithAllAccess(context.Background())
+)
+
+func TestSearchComparison(t *testing.T) {
+	suite.Run(t, new(SearchComparisonTestSuite))
+}
+
+type SearchComparisonTestSuite struct {
+	suite.Suite
+
+	mockCtrl       *gomock.Controller
+	testDB         *pgtest.TestPostgres
+	imageDatastore imageDataStore.DataStore
+
+	optionsMap search.OptionsMap
+}
+
+func (s *SearchComparisonTestSuite) SetupSuite() {
+	s.T().Setenv(env.PostgresDatastoreEnabled.EnvVar(), "true")
+
+	if !env.PostgresDatastoreEnabled.BooleanSetting() {
+		s.T().Skip("Skip postgres store tests")
+		s.T().SkipNow()
+	}
+
+	s.testDB = pgtest.ForT(s.T())
+
+	imageDS, err := imageDataStore.GetTestPostgresDataStore(s.T(), s.testDB.DB)
+	s.Require().NoError(err)
+	s.imageDatastore = imageDS
+	s.optionsMap = pkgSchema.ImagesSchema.OptionsMap
+}
+
+func (s *SearchComparisonTestSuite) TearDownSuite() {
+	//s.testDB.Teardown(s.T())
+}
+
+func compareResults(t *testing.T, matches bool, predResult *search.Result, searchResults []search.Result) {
+	assert.Equal(t, matches, len(searchResults) != 0)
+	log.Infof("Results length => %d", len(searchResults))
+	log.Infof("Results Matches => %v", searchResults[0].Matches)
+	if matches && len(searchResults) > 0 {
+		for k := range predResult.Matches {
+			log.Infof("k => %q", k)
+			sort.Strings(predResult.Matches[k])
+			sort.Strings(searchResults[0].Matches[k])
+			assert.Equal(t, predResult.Matches[k], searchResults[0].Matches[k])
+		}
+	}
+}
+
+func (s *SearchComparisonTestSuite) TestImageSearchResults() {
+	//s.T().SkipNow()
+	pgtest.SkipIfPostgresDisabled(s.T())
+
+	test := 0
+
+	cases := []struct {
+		image *storage.Image
+		query *v1.Query
+	}{
+		{
+			image: fixtures.GetImage(),
+			query: search.NewQueryBuilder().AddStringsHighlighted(search.ImageTag, "latest").ProtoQuery(),
+		},
+		{
+			image: fixtures.GetImage(),
```

Comment:
This query generates the following SQL:
```
select distinct(images.Id) as Image_Sha, image_cves.Cvss, image_cves.CveBaseInfo_Cve, 
images.LastUpdated as last_updated 
from images 
	inner join image_cve_edges on images.Id = image_cve_edges.ImageId 
	inner join image_cves on image_cve_edges.ImageCveId = image_cves.Id 
	where (image_cves.Cvss >= 5 and image_cves.CveBaseInfo_Cve is not null) 
order by images.LastUpdated asc;
```.
That results in 5 rows of being returned in the query.
```
"sha256:SHA2"	5	"CVE-2014-6200"	"2023-02-20 20:29:32.16179"
"sha256:SHA2"	5	"CVE-2014-6201"	"2023-02-20 20:29:32.16179"
"sha256:SHA2"	5	"CVE-2014-6202"	"2023-02-20 20:29:32.16179"
"sha256:SHA2"	5	"CVE-2014-6203"	"2023-02-20 20:29:32.16179"
"sha256:SHA2"	5	"CVE-2014-6204"	"2023-02-20 20:29:32.16179"
```

However, when the query is executed the results are returned as: